Assets/MRTK/Providers/LeapMotion/Editor/ConfigurationChecker/LeapMotionConfigurationChecker.cs

```diff
@@ -0,0 +1,389 @@
+                ScriptUtilities.RemoveScriptingDefinitions(BuildTargetGroup.Standalone, Definitions);
+                return false;
+            }
+        }
+
+
+        /// <summary>
+        /// Configure the Leap Motion Core assets if they are in the project.  First remove testing folders, add LeapMotion.asmdef at the
+        /// root of the core assets, and add the leap editor asmdefs.  If the core assets are not in the project, make sure the reference
+        /// in the Microsoft.MixedReality.Toolkit.Providers.LeapMotion.asmdef does not contain a ref to LeapMotion.
+        /// </summary>
+        /// <param name="isLeapInProject">Bool that determines if the Leap Motion Core assets are in the project</param>
+        private static void ConfigureLeapMotion(bool isLeapInProject)
+        {
+            FileInfo[] leapDataProviderAsmDefFile = FileUtilities.FindFilesInAssets("Microsoft.MixedReality.Toolkit.Providers.LeapMotion.asmdef");
+
+            AssemblyDefinition leapDataProviderAsmDef = AssemblyDefinition.Load(leapDataProviderAsmDefFile[0].FullName);
+
+            List<string> references = leapDataProviderAsmDef.References.ToList();
+
+            if (isLeapInProject && !references.Contains("LeapMotion"))
+            {
+                // Get the location of the Leap Core Assets relative to the root directory
+                pathDifference = GetPathDifference();
+
+                // Make sure the Leap Core Assets imported are version 4.4.0
+                bool isLeapCoreAssetsVersionSupported = LeapCoreAssetsVersionSupport();
+
+                if (isLeapCoreAssetsVersionSupported)
+                {
+                    RemoveTestingFolders();
+                    AddAndUpdateAsmDefs();
+                    AddLeapEditorAsmDefs();
+
+                    // Refresh the database because tests were removed and 10 asmdefs were added
+                    AssetDatabase.Refresh();
+                }
+                else
+                {
+                    Debug.LogError("MRTK only supports the Leap Motion Core Assets Version 4.4.0, the Leap Motion Core Assets imported are not Version 4.4.0.");
+                }
+            }
+            
+            if (!isLeapInProject && references.Contains("LeapMotion"))
+            {
+                references.Remove("LeapMotion");
+                leapDataProviderAsmDef.References = references.ToArray();
+                leapDataProviderAsmDef.Save(leapDataProviderAsmDefFile[0].FullName);
+            }
+        }
+
+        /// <summary>
+        /// Checks if the Leap Motion Core Assets are version 4.4.0.
+        /// </summary>
+        /// <returns>True, if the Leap Motion Core Assets imported are version 4.4.0.</returns>
+        private static bool LeapCoreAssetsVersionSupport()
+        {
+            string versionLeapPath = Path.Combine(Application.dataPath, pathDifference, "LeapMotion", "Core", "Version.txt");
+
+            string leapCoreAssetsVersionNumber = "4.4.0";
```

Comment:
Moving this version to the class level (i.e. a const string) would I think be better because it puts it at a place where the reader would be more likely to see it (i.e. I read class, I see oh what's this string, oh we support version 4.4 only)

Otherwise this value is somewhat buried in the functionality here.

In general if there are values that you think would be interesting for a consumer to know (or possibly configure) consider having them at a higher level.

Reply:
Moved leapCoreAssetsVersionNumber to the top of the file